tests/test_separation.py

```diff
@@ -68,6 +68,52 @@ def __unit_test_separation_function(metric):
     nose.tools.assert_raises(ValueError, metric, sources, sources)
 
 
+def __unit_test_sources_framewise_function(metric):
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        # First, test for a warning on empty audio data
+        metric(np.array([]), np.array([]), 400, 200)
+        assert len(w) == 2
+        assert issubclass(w[-1].category, UserWarning)
+        assert str(w[-1].message) == ("estimated_sources is empty, "
+                                      "should be of size (nsrc, nsample).  "
+                                      "sdr, sir, sar, and perm will all be "
+                                      "empty np.ndarrays")
+        # And that the metric returns empty arrays
+        assert np.allclose(
+            metric(np.array([]), np.array([]), 40, 20), np.array([])
+        )
+
+    # Test for error when there is a silent reference/estimated source
+    ref_sources = np.vstack((np.zeros(100),
+                             np.random.random_sample((2, 100))))
+    est_sources = np.vstack((np.zeros(100),
+                             np.random.random_sample((2, 100))))
+    nose.tools.assert_raises(ValueError, metric, ref_sources[:2],
+                             est_sources[1:], 40, 20)
+    nose.tools.assert_raises(ValueError, metric, ref_sources[1:],
+                             est_sources[:2], 40, 20)
+
+    # Test for error when shape is different
+    ref_sources = np.random.random_sample((4, 100))
+    est_sources = np.random.random_sample((3, 100))
+    nose.tools.assert_raises(ValueError, metric,
+                             ref_sources, est_sources, 40, 20)
+
+    # Test for error when too many sources are provided
+    sources = np.random.random_sample((mir_eval.separation.MAX_SOURCES*2, 400))
+    nose.tools.assert_raises(ValueError, metric, sources, sources, 40, 20)
+
+    # Test for invalid win/hop parameter detection
+    est_sources = np.random.random_sample((4, 100))
+    nose.tools.assert_raises(
+        ValueError, metric, ref_sources, est_sources, 120, 20
+    )  # test with window larger than source lengths
```

Comment:
I have no particular preference. :sunglasses: 

I think the user could be forced to pad with silence etc. just to make sure they realize what's happening, but perhaps it makes more sense to do as @bmcfee proposes (and its more consistent with librosa, mir_eval, jams' expected behavior overall :+1: ) considering that's typical in other fields (computer vision and applying image kernels, comes to mind) and just providing a warning instead that SDR might be a little too good.